assets/src/edit-story/components/library/mediaLibrary.js

```diff
@@ -30,82 +44,284 @@ const Title = styled.h3`
 
 	useEffect( loadMedia );
 
+	/**
+	 * Check if number is odd or even.
+	 *
+	 * @param {number}n Number
+	 * @return {boolean} Is even.
+	 */
+	const isEven = ( n ) => {
+		return n % 2 === 0;
+	};
+
+	/**
+	 * Generate height based on ratio of original height / width.
+	 *
+	 * @param {number}oWidth Original element width.
+	 * @param {number}oHeight Original element height.
+	 * @param {number}width Desired width.
+	 * @return {number} Relative height compared to height.
+	 */
+	const getRelativeHeight = ( oWidth, oHeight, width ) => {
+		const ratio = oWidth / width;
+		const height = Math.round( oHeight / ratio );
+		return height;
+	};
+
+	/**
+	 * Handle search term changes.
+	 *
+	 * @param {Object}evt Doc Event
+	 */
+	const onSearch = ( evt ) => {
+		setSearchTerm( evt.target.value );
+		setIsMediaLoading( false );
+		setIsMediaLoaded( false );
+	};
+
+	/**
+	 * Filter REST API calls and re-request API.
+	 *
+	 * @param {string}filter Value that is passed to rest api to filter.
+	 */
+	const onFilter = ( filter ) => {
+		if ( filter !== mediaType ) {
+			setMediaType( filter );
+			setIsMediaLoading( false );
+			setIsMediaLoaded( false );
+		}
+	};
+
 	const onClose = () => {
 		setIsMediaLoading( false );
 		setIsMediaLoaded( false );
 	};
 
-	const onSelect = ( { url, origWidth, origHeight } ) => {
-		onInsert( 'image', {
-			src: url,
-			origRatio: origWidth / origHeight,
-			width: 100,
-			height: 100,
-			x: 5,
-			y: 5,
-			rotationAngle: 0,
-		} );
+	/**
+	 * Callback of select in media picker to insert media element.
+	 *
+	 * @param {Object}attachment Attachment object from backbone media picker.
+	 */
+	const onSelect = ( attachment ) => {
+		const { url: src, mime: mimeType, width: oWidth, height: oHeight } = attachment;
+		const mediaEl = { src, mimeType, oWidth, oHeight };
+		insertMediaElement( mediaEl, DEFAULT_WIDTH );
+	};
+
+	/**
+	 * Insert element such image, video and audio into the editor.
+	 *
+	 * @param {Object}attachment Attachment object
+	 * @param {number}width      Width that element is inserted into editor.
+	 * @return {null|*}          Return onInsert or null.
+	 */
+	const insertMediaElement = ( attachment, width ) => {
+		const { src, mimeType, oWidth, oHeight } = attachment;
+		const height = getRelativeHeight( oWidth, oHeight, width );
+		const origRatio = oWidth / width;
+		if ( SUPPORTED_IMAGE_TYPES.includes( mimeType ) ) {
+			return onInsert( 'image', {
+				src,
+				width,
+				height,
+				x: 5,
+				y: 5,
+				rotationAngle: 0,
+				origRatio,
+			} );
+		} else if ( SUPPORTED_VIDEO_TYPES.includes( mimeType ) ) {
+			return onInsert( 'video', {
+				src,
+				width,
+				height,
+				x: 5,
+				y: 5,
+				rotationAngle: 0,
+				origRatio,
+				mimeType,
+			} );
+		}
+		return null;
+	};
+
+	/**
+	 * Get a formatted element for different media types.
+	 *
+	 * @param {Object}mediaEl Attachment object
+	 * @param {number}width      Width that element is inserted into editor.
```

Comment:
Even if the linter allows this right now, then for consistency, let's make sure that we have a space between the param type and the param name (e.g. `{number} width` instead of `{number}width`. Also, that the comment style would be consistent (currently one comment has a space in the front, the other multiple spaces.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3907/commits/18d1cb294e815978f5c8e0597704f70595a7bcba